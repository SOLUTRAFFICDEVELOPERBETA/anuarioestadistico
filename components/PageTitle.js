/* eslint-disable indent */
import React from 'react'
import PropTypes from 'prop-types'
import { makeStyles, useTheme, useMediaQuery, Box, Grid, Typography, Divider } from '@material-ui/core'

// Estilos del componente
const useStyles = makeStyles(theme => ({
  divider: {
    marginBottom: theme.spacing(1)
  }
}))
/**
 * @description Componente que permite integrar un titulo en una pagina
 * @param {String} title 
 * @param {String} subtitle 
 * @param {any} children 
 * @param {Object {}} actions 
 * @param {Boolean} disableDivider 
 * Consulte los elementos utilizados en {@link https://material-ui.com/ | Material-ui}
 */
const PageTitle = ( {title, subtitle, children, actions, disableDivider} ) => {
  const classes = useStyles()
  const theme = useTheme()
  const responsive = useMediaQuery(theme.breakpoints.up('md'))

  return (
    <Box marginBottom={2}>
      {actions ? (
        <Grid container spacing={1} alignItems="center" justify="space-between">
          {responsive ? (
            <React.Fragment>
              <Grid item xl={9} lg={9} md={9}>
                <Typography color="primary" variant="h4">
                  {title}
                </Typography>
              </Grid>
              <Grid item xl="auto" lg="auto" md="auto">
                {actions}
              </Grid>
              <Grid item xl={12} lg={12} md={12}>
                <Typography color="secondary" variant="subtitle1">
                  {subtitle}
                </Typography>
              </Grid>
            </React.Fragment>
          ) : (
              <React.Fragment>
                <Grid item sm={12} xs={12}>
                  <Typography color="primary" variant="h4">
                    {title}
                  </Typography>
                  <Typography color="secondary" variant="subtitle1">
                    {subtitle}
                  </Typography>
                </Grid>
                <Grid item sm={12} xs={12}>
                  {actions}
                </Grid>
              </React.Fragment>
            )}
        </Grid>
      ) : (
          <React.Fragment>
            <Typography variant="h4" color="primary">
              {title}
            </Typography>
            <Typography variant="subtitle1" color="secondary">
              {subtitle}
            </Typography>
          </React.Fragment>
        )}
      {!disableDivider && <Divider variant="middle" />}
      <div className={classes.divider}></div>
      <Typography variant="body1">
        {children}
      </Typography>
    </Box>
  )
}

PageTitle.propTypes = {
  title: PropTypes.string,
  subtitle: PropTypes.string,
  children: PropTypes.object,
  actions: PropTypes.node,
  disableDivider: PropTypes.bool
}

PageTitle.defaultProps = {
  title: '',
  subtitle: '',
  children: '',
  actions: null,
  disableDivider: false
}

export default PageTitle
