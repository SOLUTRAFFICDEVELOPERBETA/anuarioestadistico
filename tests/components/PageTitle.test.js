import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import React from 'react';
import PageTitle from '../../components/PageTitle';

describe('Pruebas en <PageTitle/>', () => {
    const actions = <button>Acciones de prueba</button>
    const wrapper = shallow(
        <PageTitle
            title="Titulo de prueba"
            subtitle="subtitulo de prueba"
            children={<p>esto es un children</p>}
            actions={actions}
            disableDivider={false}
        />
    )
    test('debe mostrarse correctamente', () => {

        expect(wrapper).toMatchSnapshot()
    })

    test('debe mostrar la informaciÃ³n correctamente', () => {
        expect(wrapper.find('WithStyles(ForwardRef(Typography))').at(0).text().trim()).toBe('Titulo de prueba')
        expect(wrapper.find('WithStyles(ForwardRef(Typography))').at(1).text().trim()).toBe('subtitulo de prueba')
        expect(wrapper.find('WithStyles(ForwardRef(Typography))').at(2).text().trim()).toBe('esto es un children')

        expect(wrapper.find('button').html()).toBe('<button>Acciones de prueba</button>');

        expect(wrapper.find('WithStyles(ForwardRef(Divider))').exists()).toBe(true);
    })

    test('debe ocultar el divider cuando disableDivider = true', () => {
        const wrapper = shallow(
            <PageTitle
                title="Titulo de prueba"
                subtitle="subtitulo de prueba"
                children={<p>esto es un children</p>}
                actions={actions}
                disableDivider={true}
            />
        )

        expect(wrapper.find('WithStyles(ForwardRef(Divider))').exists()).toBe(false);
    })
})