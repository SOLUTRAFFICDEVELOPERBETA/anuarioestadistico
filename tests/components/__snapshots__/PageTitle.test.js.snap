// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PageTitle/> debe mostrarse correctamente 1`] = `
<Styled(MuiBox)
  marginBottom={2}
>
  <WithStyles(ForwardRef(Grid))
    alignItems="center"
    container={true}
    justify="space-between"
    spacing={1}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      sm={12}
      xs={12}
    >
      <WithStyles(ForwardRef(Typography))
        color="primary"
        variant="h4"
      >
        Titulo de prueba
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        color="secondary"
        variant="subtitle1"
      >
        subtitulo de prueba
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      sm={12}
      xs={12}
    >
      <button>
        Acciones de prueba
      </button>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Divider))
    variant="middle"
  />
  <div
    className="makeStyles-divider-1"
  />
  <WithStyles(ForwardRef(Typography))
    variant="body1"
  >
    <p>
      esto es un children
    </p>
  </WithStyles(ForwardRef(Typography))>
</Styled(MuiBox)>
`;
